app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+  has_many :images, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :commented_images, through: :comments, source: :image
+  validates :username,
+  :presence => true,
+  :uniqueness => {
+    :case_sensitive => false
+  }
+  validate :validate_username
+
+  def validate_username
+    if User.where(email: username).exists?
+      errors.add(:username, :invalid)
+    end
+  end
+
+  attr_accessor :login
+
+  def self.find_first_by_auth_conditions(warden_conditions)
+    conditions = warden_conditions.dup
+    if login = conditions.delete(:login)
```

Comment:
Lint/AssignmentInCondition: Assignment in condition - you probably meant to use `==`. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)